File: cs/ipaddress/test/TestPrefix128.cs
using System;
using System.Numerics;
using System.Collections.Generic;
using System.Text;
using Xunit;
using ipaddress;

//namespace ipaddress
namespace address_test
{

  class Prefix128Test
  {
    public Dictionary<uint, BigInteger> u128_hash = new Dictionary<uint, BigInteger>();
  }


  public  class TestPrefix128 {


    Prefix128Test setup() {
      var p128t = new Prefix128Test();
      p128t.u128_hash.Add(32, BigInteger.Parse("340282366841710300949110269838224261120"));
      p128t.u128_hash.Add(64, BigInteger.Parse("340282366920938463444927863358058659840"));
      p128t.u128_hash.Add(96, BigInteger.Parse("340282366920938463463374607427473244160"));
      p128t.u128_hash.Add(126, BigInteger.Parse("340282366920938463463374607431768211452"));
      return p128t;
    }

    [Fact]
      void test_initialize() {
        Assert.True(Prefix128.create(129).isErr());
        Assert.True(Prefix128.create(64).isOk());
      }

    [Fact]
      void test_method_bits() {
        var prefix = Prefix128.create(64).unwrap();
        var str = new StringBuilder();
        for (var i = 0; i < 64; i++) {
          str.Append("1");
        }
        for (var i = 0; i < 64; i++) {
          str.Append("0");
        }
        Assert.Equal(str.ToString(), prefix.bits());
      }
    [Fact]
      void test_method_to_u32() {
        foreach (var kp in setup().u128_hash)
        {
          var num = kp.Key;
          var u128 = kp.Value;
          Assert.Equal(u128, Prefix128.create(num).unwrap().netmask());
        }

      }
  }
}

File: kotlin/src/com/adviser/ipaddress/TestPrefix128.kt
package com.adviser.ipaddress

import java.math.BigInteger
import kotlin.test.assertEquals

class Prefix128Test {
    public HashMap<Integer, BigInteger> u128_hash = HashMap<Integer, BigInteger>()
}

class TestPrefix128 {

    public fun Prefix128Test setup() {
        val p128t = Prefix128Test()
        p128t.u128_hash.put(32, BigInteger("340282366841710300949110269838224261120", 10))
        p128t.u128_hash.put(64, BigInteger("340282366920938463444927863358058659840", 10))
        p128t.u128_hash.put(96, BigInteger("340282366920938463463374607427473244160", 10))
        p128t.u128_hash.put(126, BigInteger("340282366920938463463374607431768211452", 10))
        return p128t;
    }

    @Test
    public fun test_initialize() {
        assertTrue(Prefix128.create(129).isErr());
        assertTrue(Prefix128.create(64).isOk());
    }

    @Test
    public fun test_method_bits() {
        val prefix = Prefix128.create(64).unwrap();
        var str = StringBuilder();
        for (var i = 0; i < 64; i++) {
            str.append("1");
        }
        for (var i = 0; i < 64; i++) {
            str.append("0");
        }
        assertEquals(str.toString(), prefix.bits())
    }
    @Test
    public fun test_method_to_u32() {
        setup().u128_hash.forEach[num, u128|
            assertEquals(u128, Prefix128.create(num).unwrap().netmask())
        ]
    }
}
